packages/expect/src/utils.ts

```diff
@@ -80,7 +80,14 @@ export const getPath = (
     result.traversedPath.unshift(prop);
 
     if (lastProp) {
-      result.hasEndProp = prop in object;
+      try {
+        result.hasEndProp = newObject !== undefined || prop in object;
+      } catch (e) {
+        // A primitive value usually throws TypeError: Cannot use 'in' operator
+        // Beware: 'length' in string throws, but property does exist.
+        // However, string['length'] is number, not undefined.
+        result.hasEndProp = false;
+      }
```

Comment:
In case the try/catch is only about primitives, I would prefer `... && typeof object === "object" && prop in object` over try/catch. 

In case of other situations, where `in` would throw where we really have no alternative to try/catch, then I would prefer extending the comment in the catch-clause to make it clear for the next code reader (and prevent her/him refactoring to a `typeof x === "object"` check).

Reply:
Thank you for constructive critique from viewpoint of reader. Yes, I agree that the comment is incomplete and unclear.

Here is some data. What do you suggest to improve either code or comment, or both?

* While preparing to rewrite with `typeof` I found either **object** or **function** (which was reason to add memoized test case)
* While writing this comment, I noticed yet another possibility of **Host object** (provided by the JS environment) while rereading https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#Description

Reply:
In this case, I would prefer to avoid try/catch and instead typeof-check for object/function (or maybe _negative check_ for number/string/boolean/symbol).

Reasoning: Here have to swallow/mask a thrown error completely (it wouldn't be possible to e.g. log it). In such situations it's more safe to just state all the expected exemptions than possibly hide non-`in` operator related (i.e. really unexpected) errors. If we later need some additional exemptions (e.g. for your mentioned _Host object_ or whatever) just add them later (+ fixate them with test cases).

Reply:
Yes, I agree. Thank you for clear and convincing reasoning.

Reply:
Okay, great! And thanks for fixing this issue!

Reply:
This needs https://github.com/facebook/jest/pull/7708...

Reply:
Merged that, could we use `import {isPrimitive} from 'jest-get-type';`? And if that's insufficient, fix it so it is?

Reply:
Thank you. I just pushed another refactoring commit :)